app/src/main/java/com/example/ncov19traking/data/NCoVRepository.kt

```diff
@@ -74,7 +78,6 @@ class NCoVRepository(nCoVDataBase: NCoVDataBase) {
         return countryDao.load()
     }
 
-
     suspend fun getHistoricalCountryData() = NCoVApiClient.nCoVApi.getHistoricalDataByCountry()
```

Comment:
you are missing the Dagger optimization for Retrofit. the `NCoVApiClient` class should not exist, you should provide the Retrofit Interceptor and also provide the Retrofit client instance and inject it where you need it, and in the classes you are going to use you should provide `NCoVApi`, no class should know about the retrofit client at all. Investigate in internet how to inject retrofit using dagger.